pkg/model/iam/iam_builder.go

```diff
@@ -343,6 +343,7 @@ func (b *PolicyBuilder) AddS3Permissions(p *Policy) (*Policy, error) {
 							strings.Join([]string{b.IAMPrefix(), ":s3:::", iamS3Path, "/instancegroup/*"}, ""),
 							strings.Join([]string{b.IAMPrefix(), ":s3:::", iamS3Path, "/pki/issued/*"}, ""),
 							strings.Join([]string{b.IAMPrefix(), ":s3:::", iamS3Path, "/pki/private/kube-proxy/*"}, ""),
+							strings.Join([]string{b.IAMPrefix(), ":s3:::", iamS3Path, "/pki/private/kube-router/*"}, ""),
```

Comment:
We want to only add this if kube-router is being used

Reply:
See https://github.com/kubernetes/kops/pull/3778  just make sure you check for nil on the test, as they did not

Reply:
Sure, will get this updated